src/main.js

```diff
@@ -1,39 +1,16 @@
 Object.assign(extension, {
-  searchboxSelector: '#searchform input[name=q]',
+  searchEngine: getSearchEngine(),
 
-  init() {
-    if (!/^(www|encrypted)\.google\./.test(window.location.hostname)) {
-      return;
-    }
-
-    const loadOptions = this.options.load();
+   init(loadOptions) {
     // Don't initialize results navigation on image search, since it doesn't work
     // there.
     if (!/[?&]tbm=isch(&|$)/.test(location.search)) {
```

Comment:
This part looks specific to Google. Perhaps you should also make the `init` function part of the search engine object.

Reply:
You're right. Actually, search engine objects didn't have `init` function (added in ca79773). It will help to disable results init on non compatible page depending on search engine.